### PR TITLE
Extract payloads apiserver parts

### PR DESCRIPTION
**Pull request by mjs**
## Description of change

This PR moves the apiserver components of the "payloads" feature from the top level `payload` package to under the `apiserver` package where the rest of the API is defined. It is part of a larger effort to clean up the apiserver in preparation for a major new feature.

## QA steps

There should be no externally visible changes. In order to check that a regression hasn't been introduced a payload using charm was deployed:

```
$ juju bootstrap lxd
$ juju deploy ~/canonical/repository/payloadsfoo
# wait until it deploys
$ juju list-payloads
[Unit Payloads]
Unit            Machine  Payload class  Status   Type    Id   Tags                                   
payloadsfoo/0   1        foo            running  mocker  Wed  Mar 8 19:28:26 UTC 2017
```

This ensures that both the unit facade (which is used to set payloads) and the payloads facade (which is used to retrieve payloads) are working.

## Documentation changes

No externally visible changes.

## Bug reference

N.A.

### CODE REVIEW

File: apiserver/params/payloads.go [hunk 1]
```diff
@@ -1,17 +1,16 @@
-// Copyright 2015 Canonical Ltd.
+// Copyright 2017 Canonical Ltd.
```
Comment (axw) [inline apiserver/params/payloads.go]: 2015-2017 ?

Comment (mjs) [inline apiserver/params/payloads.go]: *shrug* - it's been moved from elsewhere so it's new here.

File: payload/api/helpers.go [hunk 2]
```diff
@@ -8,12 +8,25 @@ import (
 	"gopkg.in/juju/charm.v6-unstable"
 	"gopkg.in/juju/names.v2"
 
+	"github.com/juju/juju/apiserver/params"
 	"github.com/juju/juju/payload"
 )
 
+// API2ID converts the given payload tag string into a payload ID.
```
Comment (mjs) [inline payload/api/helpers.go]: Done

**Comment by mjs** [comment]
$$merge$$

**Comment by mjs** [comment]
$$merge$$

**Comment by mjs** [comment]
$$merge$$

**Pull request merged**